import.go

```diff
@@ -109,54 +101,131 @@ func importYamlWithTemplateFunc(r io.Reader, tmplFunc TemplateFunc) ([]*YamlSect
 		}}
 	}
 	for i := range chunks {
-		vp := viper.New()
-		vp.SetConfigType("yaml")
+		//Each chunk is a document, needs to be filtered so the consumer gets the sections they want
+		// then run the supplied template, then Vipered
+		// this avoids running templates on unrelated sections that we may not have the data for
+
+		//extract section
 		var b []byte
 		if i < len(chunks)-1 {
 			b = data[chunks[i][1]:chunks[i+1][0]]
 		} else {
 			b = data[chunks[i][1]:]
 		}
-		if err := vp.ReadConfig(bytes.NewBuffer(b)); err != nil {
-			fmt.Printf("---\n%v\n", string(b))
-			return nil, errors.Wrap(err, "failed to import yaml section")
-		}
-		sections = append(sections, &YamlSection{Bytes: b, Viper: vp})
+
+		//save completed section
+		sections = append(sections, &YamlSection{
+			Bytes:         b,
+			OriginalBytes: b,
+		})
 	}
 	return sections, nil
 }
 
-func importYaml(r io.Reader) ([]*YamlSection, error) {
-	sections := []*YamlSection{}
-	buf := new(bytes.Buffer)
-	_, err := buf.ReadFrom(r)
+func Filter(in []*YamlSection, filters []string) ([]*YamlSection, error) {
+	ret := []*YamlSection{}
```

Comment:
It is a matter of personal taste to some extend but I do like to name variables explicitly to help readability and reduce cognitive complexity. We can discuss standards around this some other time if you want.

```suggestion
	sections := []*YamlSection{}
```

Reply:
I don't have a strong disagreement, I do subscribe to the "shorter viable names for shorter functions", but yours does not look bad to me. So the question is should we resolve it in this PR? Or do it as a separate PR..

Reply:
This PR would be good. I agree, I try to keep them short, and I also follow the Go best practices. So for things like io.Reader (r), and function receivers, simple names are good. Strings are just particularly ambiguous.

Reply:
Sound argument. Will adapt the more descriptive style and fix this one.